### ISSUE TITLE
Convert .eslintrc to .eslintrc.js and add comments

### ISSUE THREAD

**Issue by kylemh** [OperationCode/front-end#481]
# Feature request
**Is your feature request related to a problem? Please describe.**
As a beginner-friendly project, we provide a lot of opinionation and good-practice guidance via our larger ESLint configuration. So large, in fact, that I have trouble quickly scanning all that it's doing 😅

**Describe the solution you'd like**
By converting the file to `.js`, sorting the rules first by plugin name and then alphabetically by rule name, and by adding comments and newlines between each plugin grouping, we'll be setting me up for success in managing these rules more quickly in the future.

To summarize:
- Add `.js` to the suffix of the file
- Add `module.exports = ` to the very first character of the file
- Run `yarn lint` to ensure the command still works (you should see no file changes, or only file changes to the `.eslintrc.js` file (if you did something that required fixing).
- Group rules first by plugin name:
```
// Vanilla ESLint Rules
"comma-dangle": ["error", "only-multiline"],
/* others */

// Jest Plugin Rules
"jest/consistent-test-it": ["error", { "fn": "it", "withinDescribe": "it" }],
/* others */

// JSX A11y Plugin Rules
// etc.

// React Plugin Rules
// etc.
```

- Sort each grouping alphabetically
- Add comments as shown in the example above for each grouping:
```
// Vanilla ESLint Rules
// Jest Plugin Rules
// etc...
```

**Comment by kylemh** [comment]
I am assigning myself to this issue as a placeholder for ryosts’s work. Thank you, and good luck! 😁 

**Please note:**
If no activity is shared within two weeks, I will remove my assignment. At that point, the issue will be open for anybody to take.

**Comment by kylemh** [comment]
Closed in #482
